core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java

```diff
@@ -40,14 +44,21 @@ protected void preBuild(List<String> buildArgs) throws InterruptedException, IOE
     }
 
     @Override
-    protected void postBuild() throws InterruptedException, IOException {
-        String[] copyCommand = new String[] { containerRuntime.getExecutableName(), "cp",
-                containerId + ":" + NativeImageBuildStep.CONTAINER_BUILD_VOLUME_PATH + "/" + nativeImageName, outputPath };
-        Process copyProcess = new ProcessBuilder(copyCommand).start();
-        copyProcess.waitFor();
+    protected void postBuild(Logger log) throws InterruptedException, IOException {
+        copy(log, nativeImageName);
+        copy(log, "sources");
```

Comment:
Copying `sources` is done whether debug is enabled or not. Does it work in both cases?

Reply:
It does work (it ignores the failure and doesn't print any error). Now that you mention it though I am not sure we are OK with that. @geoand what do you think? Should we check the exit code of the `docker cp` and report failures to the user?

Reply:
We can at least log it if the user has requested the sources to be present

Reply:
I did some more work on logging (https://github.com/quarkusio/quarkus/pull/15288/commits/b2904f0731a2d47b8ba9d5e6380b62d0954aa313)

Reply:
Note: I might have made it a bit to verbose for INFO:

```
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildRunner] podman run --env LANG=C --rm quay.io/quarkus/ubi-quarkus-native-image:21.0.0-java11 --version
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Running Quarkus native-image plugin on GraalVM Version 21.0.0 (Java Version 11.0.10+8-jvmci-21.0-b06)
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildRemoteContainerRunner] podman create --env LANG=C quay.io/quarkus/ubi-quarkus-native-image:21.0.0-java11 -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-DCoordinatorEnvironmentBean.transactionStatusManagerEnable=false -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -J-Dcom.sun.xml.bind.v2.bytecode.ClassTailor.noOptimize=true -J-Drx.unsafe-disable=true -J-Duser.language=en -J-Duser.country=IE -J-Dfile.encoding=UTF-8 -H:ReflectionConfigurationFiles=reflection-config.json --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy\$BySpaceAndTime -H:+JNI -H:+AllowFoldMethods -jar quarkus-integration-test-main-999-SNAPSHOT-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:+AddAllCharsets -H:EnableURLProtocols=http --no-server -H:-UseServiceLoaderFeature -H:+StackTrace quarkus-integration-test-main-999-SNAPSHOT-runner
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildRunner] podman cp /home/zakkak/code/quarkus/integration-tests/main/target/quarkus-integration-test-main-999-SNAPSHOT-native-image-source-jar/. 9191309dddf8eb2e2e5e7c77f61e8fcfb929c86033c95161f8d9c50bebbebe21:/project
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildRunner] podman start --attach 9191309dddf8eb2e2e5e7c77f61e8fcfb929c86033c95161f8d9c50bebbebe21
[quarkus-integration-test-main-999-SNAPSHOT-runner:20]    classlist:  16,959.76 ms,  1.70 GB
[quarkus-integration-test-main-999-SNAPSHOT-runner:20]        (cap):     899.37 ms,  1.70 GB
[quarkus-integration-test-main-999-SNAPSHOT-runner:20]        setup:   4,376.07 ms,  1.70 GB
13:38:41,492 INFO  [org.hib.val.int.uti.Version] HV000001: Hibernate Validator 6.2.0.Final
13:38:42,027 INFO  [org.hib.Version] HHH000412: Hibernate ORM core version 5.4.28.Final
13:38:42,047 INFO  [org.hib.ann.com.Version] HCANN000001: Hibernate Commons Annotations {5.1.2.Final}
13:38:42,162 INFO  [org.hib.dia.Dialect] HHH000400: Using dialect: org.hibernate.dialect.PostgreSQL95Dialect
13:38:43,293 WARNING [org.apa.tik.par.PDFParser] J2KImageReader not loaded. JPEG2000 files will not be processed.
See https://pdfbox.apache.org/2.0/dependencies.html#jai-image-io
for optional dependencies.

13:39:26,308 INFO  [org.jbo.threads] JBoss Threads version 3.2.0.Final
[quarkus-integration-test-main-999-SNAPSHOT-runner:20]     (clinit):   1,458.28 ms,  6.10 GB
[quarkus-integration-test-main-999-SNAPSHOT-runner:20]   (typeflow):  44,649.57 ms,  6.10 GB
[quarkus-integration-test-main-999-SNAPSHOT-runner:20]    (objects):  51,428.66 ms,  6.10 GB
[quarkus-integration-test-main-999-SNAPSHOT-runner:20]   (features):   2,686.02 ms,  6.10 GB
[quarkus-integration-test-main-999-SNAPSHOT-runner:20]     analysis: 104,864.86 ms,  6.10 GB
[quarkus-integration-test-main-999-SNAPSHOT-runner:20]     universe:   4,678.23 ms,  6.10 GB
[quarkus-integration-test-main-999-SNAPSHOT-runner:20]      (parse):  11,138.94 ms,  6.15 GB
[quarkus-integration-test-main-999-SNAPSHOT-runner:20]     (inline):  18,534.45 ms,  8.03 GB
[quarkus-integration-test-main-999-SNAPSHOT-runner:20]    (compile):  65,737.04 ms,  7.39 GB
[quarkus-integration-test-main-999-SNAPSHOT-runner:20]      compile:  99,857.19 ms,  7.39 GB
[quarkus-integration-test-main-999-SNAPSHOT-runner:20]        image:  11,505.31 ms,  7.23 GB
[quarkus-integration-test-main-999-SNAPSHOT-runner:20]        write:   1,392.41 ms,  7.23 GB
[quarkus-integration-test-main-999-SNAPSHOT-runner:20]      [total]: 244,771.21 ms,  7.23 GB
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildRunner] podman cp 9191309dddf8eb2e2e5e7c77f61e8fcfb929c86033c95161f8d9c50bebbebe21:/project/quarkus-integration-test-main-999-SNAPSHOT-runner /home/zakkak/code/quarkus/integration-tests/main/target/quarkus-integration-test-main-999-SNAPSHOT-native-image-source-jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildRunner] podman container rm --volumes 9191309dddf8eb2e2e5e7c77f61e8fcfb929c86033c95161f8d9c50bebbebe21
```

Please advise which parts should be moved to DEBUG

Reply:
I don't think the version command needs to be INFO

Reply:
Fixed